gdx/src/com/badlogic/gdx/scenes/scene2d/ui/Skin.java

```diff
@@ -427,6 +428,49 @@ protected Json getJsonLoader (final FileHandle skinFile) {
 					return get(jsonData.asString(), type);
 				return super.readValue(type, elementType, jsonData);
 			}
+
+			static final String CASCADE_PARENT_TAG = "parent";
+
+			protected boolean isNonexistantFieldKnown (Class type, String fieldName) {
+				return fieldName.equals(CASCADE_PARENT_TAG);
+			}
+
+			public void readFields (Object object, JsonValue jsonMap) {

```

Comment:
I wonder if this can be simplified. Eg, if we assume `parent` is never a valid field name AND we have some way to copy the parent, then we could do this in Json `readValue`:

```
if (jsonData.isObject()) {
	String parentName = jsonData.getString("parent", null);
	if (parentName != null) {
		Object parent = findParent(parentName);
		Object copy = copy(parent);
		json.readFields(copy, jsonData);
		return copy;
	}
}
```

I like that this doesn't require exposing more of the Json class. For `copy()`, a hack would be to serialize to JSON and read it back as a copy. It would be better as a new Json function, since copying is very close to serialization (Kryo does both). It's also common to need a copy of a style and we have copy constructors for that, but it's probably awkward to use those here.

Reply:
Having used this in my own UI-heavy project for the past year and a half or so, I'd say the cascading feature would be roughly half as useful if we remove the ability for a parent style to be an ancestor class (for example using a ButtonStyle as the parent of a TextButtonStyle).

I suppose Json could have one more `copy` method where you specify the subclass of parent you want the "copy" to be. But is that getting too specialized? I can tinker with this in the next day or two.

Reply:
@cypherdare see what you think about dfc9f4789171e03f468faaa81343795306569c34.